.github/workflows/pythonpackage.yml

```diff
@@ -0,0 +1,64 @@
+name: Python package
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 3 # We need this limit or else the tests fail
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Upgrade all installed pip packages (setuptools, pip, etc.)
+      run: pip list --outdated --format=freeze | grep -v '^\-e' | cut -d = -f 1  | xargs -n1 pip install -U
+    - name: Install project dependencies
+      run: python setup.py install
+    - name: Install test dependencies
+      run: pip install black flake8 pydocstyle sphinx sphinx_rtd_theme
+    - name: Check code formatting with black
+      run: black --check --verbose .
+    - name: Lint with flake8
+      run: flake8 --exclude=.eggs,build,docs
+    - name: Check docstyle with pydocstyle
+      run: pydocstyle praw
+    - name: Build Sphinx files
+      run: sphinx-build -W docs/ /tmp/foo
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 4 # We need this limit or else the tests fail
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
```

Comment:
I like linting on each system, but I don't think it's necessary to run the tests on each. I guess the question is how long does it take for a commit's tests to complete?

Reply:
For me it takes about 6-7 mins.

Reply:
I think tests are necessary to make sure behaviors on linux match windows behaviors.